File: src/de/hsh/daniel/model/Game.java
package de.hsh.daniel.model;

import de.hsh.daniel.model.board.Board;
import de.hsh.daniel.model.board.GUIBoard;
import javafx.scene.canvas.Canvas;


/**
 * THis is the game started
 */
public final class Game {

    private GUIBoard board;

    public final void render( Canvas gameCanvas, final int fps ) {
        board.draw(gameCanvas);
    }

    /**
    Sets up Board on canvas and creates two Player objects
     */
    public final void initialize( Canvas gameCanvas ) {
        board = new GUIBoard();

        gameCanvas.setOnMouseClicked(e -> {
            board.onMouseClick(e.getX(), e.getY());
        });
    }

    public Board getGUIBoard() { return board.getBoard(); }


   // public Board getBoard() { return this.board; }
}

File: Games/Daniel/src/de/hsh/daniel/model/Game.kt
package de.hsh.daniel.model

import de.hsh.daniel.model.board.Board
import de.hsh.daniel.model.board.GUIBoard
import javafx.event.EventHandler
import javafx.scene.canvas.Canvas
import javafx.scene.input.MouseEvent

/**
 * THis is the game started
 */
class Game(gameCanvas: Canvas) {
    private var board: GUIBoard = GUIBoard()
    fun render(gameCanvas: Canvas) = board.draw(gameCanvas)

    init {
        gameCanvas.onMouseClicked = EventHandler { e: MouseEvent ->
            board.onMouseClick(e.x, e.y)
        }
    }

    /**
     * *****************************************************************
     * *                                                               *
     * *                       GETTERS & SETTERS                       *
     * *                                                               *
     * *****************************************************************
     *
     */
    val gUIBoard: Board
        get() = board.board
}
